database/apid/src/main/java/com/example/javaopencommerce/item/ItemEntity.java

```diff
@@ -20,6 +24,21 @@
 @EqualsAndHashCode
 public class ItemEntity {
 
+    static ItemEntity fromSnapshot(ItemSnapshot itemSnapshot) {
+        ImageEntity image = ImageEntity.builder()
+            .id(itemSnapshot.getImage().getId())
```

Comment:
it's not null safe operation. Consider extracting image to variable:
```suggestion
ImageSnapshot img = Optional.ofNullable(itemSnapshot).map(ItemSnapshot::getImage).orElse(ImageSnapshot.builder.build();
ImageEntity image = ImageEntity.builder()
.id(img.getId())
```

Reply:
Fixed!